From ac4e3d4d6e12dfc4811a33ff3bda86c5ae73ef13 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Wed, 11 Jul 2018 20:44:36 -0700
Subject: [PATCH 2/5] Set EUID instead of loginuid for swdb history

The getloginuid() function tries to read from /proc/self/loginuid, but
this file never exists on Clear Linux. When the file does not exist,
None is returned, but the libdnf C++ API expects an unsigned integer.

Simply pass the effective UID here instead of loginuid.

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 dnf/db/history.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dnf/db/history.py b/dnf/db/history.py
index f4158ec..cc4e1d9 100644
--- a/dnf/db/history.py
+++ b/dnf/db/history.py
@@ -445,7 +445,7 @@ class SwdbInterface(object):
             int(calendar.timegm(time.gmtime())),
             str(rpmdb_version),
             cmdline or "",
-            int(misc.getloginuid())
+            os.geteuid()
             )
         self.swdb.setReleasever(self.releasever)
         self._tid = tid
-- 
2.19.1

